File: backend/src/main/java/com/drinkster/dto/PlayerDto.java
package com.drinkster.dto;

import com.drinkster.model.Player;

public record PlayerDto(
        String id,
        String name,
        String sex,
        boolean isAdmin,
        boolean isReady,
        boolean isPlaying
) {
    public static PlayerDto fromPlayer(Player player) {
        return new PlayerDto(
                player.getId().toString(),
                player.getName(),
                player.getSex().toString(),
                player.isAdmin(),
                player.isReady(),
                player.isPlaying()
        );
    }
}

File: frontend/src/app/models/dto/player.dto.ts
import {ChallengeDto} from "./Challenge.dto";

export interface PlayerDto {
  id: string;
  name: string;
  sex: string;
  isAdmin: boolean;
  isReady: boolean;
  isPlaying?: boolean

}

export interface PlayerTurnDto {
  player: PlayerDto;
  challenge: ChallengeDto
  affectedPlayers: PlayerDto[];
}
